### ISSUE TITLE
Certificate can’t be blank error when posting comment

### ISSUE THREAD

**Issue by leewynne** [forem/forem#14613]
**Describe the bug**

On a number of Forems when I have attempted to add a comment I receive an 'error posting comment'  bug with the detail 'there was a error with your markdown, certificate can't be blank'. However, when you hit cancel, you can see you comment has actually been successful submitted.

**To Reproduce**

The bug seems sporadic. Sometimes I receive it forems and sometimes I don't. The most recent example is Nick's vscode tips forem, this was in the forem mobile app on iOS.

Some members of thismmalife have also reported this when on Chrome on PC (but couldn't verify of it was in a private tab or not)

**Expected behavior**

Comment is successfully submitted.

**Screenshots**

![image](https://user-images.githubusercontent.com/12694270/131218670-b2f34d52-49f0-48a9-8a92-12ab95118ff4.jpeg)

**Comment by djuber** [comment]
if the comment is being created/saved correctly (and shows on a page refresh, error notwithstanding)- it's almost certainly some of the _other_ actions processed [here](https://github.com/forem/forem/blob/main/app/controllers/comments_controller.rb#L66-L83) that are producing the error shown during comment creation.

It looks like [Rpush::Client::ActiveModel::Apns2::App](https://github.com/rpush/rpush/blob/master/lib/rpush/client/active_model/apns2/app.rb#L9) (from an installed gem, used for mobile app notifications) adds a model validation for `certificate`, and is possibly not always set up correctly leading to this error.

I'll ping @nickytonline, since he'll  have server logs for the vscodetips forem to produce a backtrace from the error (and better knowledge about the configuration state of that forem), but I would guess the [`Notifications::NewComment::Send`](https://github.com/forem/forem/blob/main/app/services/notifications/new_comment/send.rb#L37-L49) might be involved - this I think comes from the [comment controller](https://github.com/forem/forem/blob/main/app/controllers/comments_controller.rb#L73)'s sending of new comment notifications

```ruby
      Notification.send_new_comment_notifications_without_delay(@comment)
```

which in turn calls the Send notification action

```ruby
      Notifications::NewComment::Send.call(comment)
 ```

which calls Push Notifications (for mobile).

The certificate for the ios app might be the one in the `ConsumerApps::RpushAppQuery#recreate_ios_app!`, this ultimately comes from the `RPUSH_IOS_PEM` environvment variable (which I'm guessing is set to an empty string, causing the `presence` validation to fail.

**Comment by nickytonline** [comment]
I can confirm it happens on the desktop as well.

![image](https://user-images.githubusercontent.com/833231/131236306-74e32713-ba70-44c4-ab98-c9a554bbbb92.png)

**Comment by citizen428** [comment]
Might also be worth to ping @fdoxyz, he probably knows the most about our push notification implementation.

**Comment by fdoxyz** [comment]
Thanks @leewynne & @nickytonline for reporting/confirming this! Also, you're spot on @djuber.

The trickiness here (reason why it was only "sporadically" bubbling up) was that this only happens when someone has devices registered to get PNs delivered. At this time, this only happens when someone used the iOS app to browse a selfhosted Forem. If user X wrote a Post and user Y replies, no error would appear if X didn't use the iOS app. But if user Y uses the app with it's device registered to receive PNs and user X replies to their comment, then this error comes up because a PN delivery is attempted to user X.

Long story short: I have a PR up to fix this (working on some tests to avoid regressions).
